src/engine/world/room.cpp

```diff
@@ -584,12 +582,47 @@ void Room::collectShaderLights(size_t depth)
   lightsBuffer->setData(bufferLights, gl::api::BufferUsage::StaticDraw);
 }
 
-void Room::createParticleMesh(const std::string& label,
+void Room::regenerateDust(gslu::nn_shared<render::scene::Material> dustMaterial,
+                          bool isDustEnabled,
+                          uint8_t dustResolutionDivisor)
+{
+  
+  if(!isDustEnabled)
+  {
+    dust = nullptr;
+    return;
+  }
+
+  auto search = dustCache.find(dustResolutionDivisor);
+  if(search != dustCache.end())
+  {
+     dust = search->second;
+  }
+  else
+  {  
+    glm::vec3 min{std::numeric_limits<float>::max()};
+    glm::vec3 max{std::numeric_limits<float>::lowest()};
+    for(const auto& v : dustVerticies)
+    {
+      const auto vv = v.position.toRenderSystem();
+      min = glm::min(min, vv);
+      max = glm::max(max, vv);
+    }
+    auto dustNode = createParticleMesh(
+      dustLabel, min, max, dustMaterial, dustResolutionDivisor);
+      dustCache.emplace(dustResolutionDivisor, dustNode);    
+      dust = dustNode;
+  }
+  
+}
+
+std::shared_ptr<render::scene::Node> Room::createParticleMesh(const std::string& label,
                               const glm::vec3& min,
                               const glm::vec3& max,
-                              render::scene::MaterialManager& materialManager)
+                              gslu::nn_shared<render::scene::Material> dustMaterial,
+                              uint8_t resolutionDivisor)
 {
-  static const constexpr auto Resolution = (1_sectors / 12).cast<float>().get();
+  static const auto resolution = (1_sectors / 12).cast<float>().get() * resolutionDivisor;
```

Comment:
```suggestion
  static const auto resolution = (1_sectors / 12).cast<float>().get() * static_cast<float>(resolutionDivisor);
```

The calculation is not entirely fitting, though. It is correct to increase the dust particle grid size, but I think it's more intuitive if the divisor would affect the density, which means you need to calculate the grid size with the cubic root of `resolutionDivisor`. This would also mean that `resolutionDivisor` is wrong and should be `densityDivisor`.

Reply:
Is this the correct formula? 
`resolution = (1_sectors / 12).cast<float>().get() * static_cast<float>(cbrt(densityDivisor));`